src/scripts/main.js

```diff
@@ -1,5 +1,25 @@
 'use strict';
 
 document.addEventListener('click', (e) => {
-  // write code here
+  const spider = document.querySelector('.spider');
+  const wall = document.querySelector('.wall');
+
+  if (e.target !== wall) {
+    return;
+  }
+
+  const posX = e.clientX;
+  const posY = e.clientY;
+
+  const wallRect = wall.getBoundingClientRect();
+  const spiderRect = spider.getBoundingClientRect();
+
+  let x = posX - wallRect.left - spiderRect.width / 2 - 10;
+  let y = posY - wallRect.top - spiderRect.height / 2 - 10;
+
+  x = Math.max(0, Math.min(x, wallRect.width - spiderRect.width - 20));
+  y = Math.max(0, Math.min(y, wallRect.height - spiderRect.height - 20));
```

Comment:
The hardcoded value of 20 used in the boundary calculations should be reviewed. If this value is meant to account for padding or other layout considerations, it might be better to define it as a constant or variable to improve code readability and maintainability.